h/form.py

```diff
@@ -15,16 +19,53 @@
 )
 
 
-def init():
-    from deform import Form
-    from deform_jinja2 import jinja2_renderer_factory
-    from deform_jinja2.translator import PyramidTranslator
+class Jinja2Renderer(object):
+    """
+    An alternate Deform renderer that uses Jinja2 to render templates.
 
-    renderer = jinja2_renderer_factory(search_paths=SEARCH_PATHS,
-                                       translator=PyramidTranslator())
+    We supply our own Jinja2 renderer because deform_jinja2's one doesn't
+    support autoescaping.
+    """
 
-    Form.set_default_renderer(renderer)
+    def __init__(self, base_env):
+        """
+        Return a new callable Jinja2Renderer object.
 
+        :param base_env: the initial Jinja2 environment that this renderer's
+            Jinja2 environment will be created as an overlay of
+        :type base_env: :py:class:`jinja2.Environment`
+        """
+        self._base_env = base_env
+        self._env = None
 
-def includeme(_):
-    init()
+    def _get_env(self):
+        """
+        Return this renderer's Jinja2 environment.
+
+        Initialize the environment if it hasn't been initialized already.
+        """
+        if not self._env:
+            resolver = AssetResolver()
+            searchpath = [resolver.resolve(path).abspath()
+                          for path in SEARCH_PATHS]
+            loader = pyramid_jinja2.SmartAssetSpecLoader(searchpath)
+            self._env = self._base_env.overlay(autoescape=True, loader=loader)
+        return self._env
+
+    def __call__(self, template_name, **kwargs):
+        """Return the given template rendered with the given keyword args."""
+        if not template_name.endswith('.jinja2'):
+            template_name += '.jinja2'
+
+        template = self._get_env().get_template(template_name)
+
+        return jinja2.Markup(template.render(**kwargs))
+
+
+def initialize_deform_renderer(config):
+    deform.Form.set_default_renderer(
```

Comment:
It's a small thing, but if you find yourself breaking a line in situations like this, then it's probably a good sign that an extra variable or two would provide some additional clarity:

```python
renderer = Jinja2Renderer(config.get_jinja2_environment())
deform.Form.set_default_renderer(renderer)
```